plugins/db_manager.py

```diff
@@ -174,6 +175,62 @@ async def acl_show(ctx: Context, acl: str) -> None:
 
     await ctx.send(format("{!b:yaml}", yaml.dump(data)))
 
+@acl_command.command("find")
+@privileged
+async def acl_find(ctx: Context, type: str, id: int) -> None:
+    """ Find all ACLs that directly reference the given user, role, channel, or category ID"""
+
+    if ctx.guild is None:
+        raise UserError("Can only be used in a guild")
+
+    if type not in ["role", "channel", "category", "user"]:
+        raise UserError("Invalid type, must be one of `user`, `role`, `channel`, or `category`")
+    if type == "role" and ctx.guild.get_role(id) is None:
+        raise UserError(format("No role with ID: {!i}", id))
+    if type in ("channel", "category") and ctx.guild.get_channel(id) is None:
+        raise UserError(format("No channel with ID: {!i}", id))
+    if type == "user" and ctx.guild.get_member(id) is None:
+        raise UserError(format("No user with ID: {!i}", id))
+    else:
+        user = ctx.guild.get_member(id)
+
+    all_acls: Set[str] = set(name for ty, name in bot.acl.conf if ty == "acl")
+    matched_acls: Set[str] = set(acl for acl in all_acls if find_in_acl(bot.acl.conf["acl", acl], type, id))
+    indirect_acls: Set[Tuple[str,int]] = set()
+    output = ""
+
+    if user is not None:
+        # matches via roles
+        indirect_acls = set((acl, role.id) 
+                            for acl in all_acls 
+                            for role in user.roles 
+                            if find_in_acl(bot.acl.conf["acl", acl], "role", role.id))
+
+    if len(matched_acls):
+        output += format("ACLs directly referencing {} {!i}: {}\n", type, id, 
+                        ", ".join(format("{!i}", acl) for acl in matched_acls))
+    if len(indirect_acls):
+        output += format("ACLs indirectly referencing {} {!i}:\n- {}", type, id,
+                        "\n- ".join(format("{!i} via <@&{}>", acl, role) for acl, role in indirect_acls))
```

Comment:
instead of `"\n- " + "\n- ".join(... for ...)` why not `"".join("\n- " + ... for ...)`

Reply:
To mention a role use `{!M}`